datafusion/functions-aggregate/src/bit_and_or_xor.rs

```diff
@@ -249,23 +253,22 @@ impl AggregateUDFImpl for BitwiseOperation {
     }
 
     fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
-        let arg_type = &arg_types[0];
-        if !arg_type.is_integer() {
-            return exec_err!(
-                "[return_type] {} not supported for {}",
-                self.name(),
-                arg_type
-            );
-        }
-        Ok(arg_type.clone())
+        Ok(arg_types[0].clone())
     }
 
     fn accumulator(&self, acc_args: AccumulatorArgs) -> Result<Box<dyn Accumulator>> {
         downcast_bitwise_accumulator!(acc_args, self.operation, acc_args.is_distinct)
     }
 
     fn state_fields(&self, args: StateFieldsArgs) -> Result<Vec<FieldRef>> {
-        if self.operation == BitwiseOperationType::Xor && args.is_distinct {
+        if args.input_fields[0].data_type().is_null() {
```

Comment:
Needing to change state_fields is not desirable, but until we can make accumulators own state_fields (instead of the function) this will have to do

- See: https://github.com/apache/datafusion/issues/14701#issuecomment-3419742933